sceptre/connection_manager.py

```diff
@@ -7,14 +7,55 @@
 Boto3 calls.
 """
 
-
+import functools
 import logging
 import threading
+import time
 
 import boto3
+from botocore.exceptions import ClientError
 
 from .helpers import mask_key
-from .helpers import exponential_backoff
+from .exceptions import RetryLimitExceededError
+
+
+def _retry(func):
```

Comment:
For separation of concerns - the `call` code makes boto calls, while the retry retries stuff. It also makes it easier to test - we can test the retry functionality without instantiating a boto session.

It's implemented as a decorator because we need to run code before and after the decorated function is run.